qutip/solver/flimesolve.py

```diff
@@ -0,0 +1,705 @@
+            basis.
+
+        args : dict,optional {None}
+            Not supported
+
+        .. note::
+            The state must be initialized first by calling ``start`` or
+            ``run``. If ``run`` is called,``step`` will continue from the last
+            time and state obtained.
+        """
+        if args:
+            raise ValueError(
+                "FMESolver cannot update arguments")
+        state = super().step(t)
+        if not floquet:
+            state = self.floquet_basis.from_floquet_basis(
+                state, t)
+        elif copy:
+            state = state.copy()
+        return state
+
+    def run(self, state00, taulist, *, floquet=False, args=None, e_ops=None,):
+        """
+        Calculate the evolution of the quantum system.
+
+        For a ``state0`` at time ``tlist[0]`` do the evolution as directed by
+        ``rhs`` and for each time in ``tlist`` store the state and/or
+        expectation values in a :class:`Result`. The evolution method and
+        stored results are determined by ``options``.
+
+        Parameters
+        ----------
+        state0 : :class:`Qobj`
+            Initial state of the evolution.
+
+        tlist : list of double
+            Time for which to save the results (state and/or expect) of the
+            evolution. The first element of the list is the initial time of the
+            evolution. Each times of the list must be increasing,but does not
+            need to be uniformy distributed.
+
+        floquet : bool,optional {False}
+            Whether the initial state in the floquet basis or laboratory basis.
+
+        args : dict,optional {None}
+            Not supported
+
+        e_ops : list {None}
+            List of Qobj,QobjEvo or callable to compute the expectation
+            values. Function[s] must have the signature
+            f(t : float,state : Qobj) -> expect.
+
+        Returns
+        -------
+        results : :class:`qutip.solver.FloquetResult`
+            Results of the evolution. States and/or expect will be saved. You
+            can control the saved data in the options.
+        """
+
+        if args:
+            raise ValueError(
+                "FLiMESolver cannot update arguments")
+        if not floquet:
+            state0 = self.floquet_basis.to_floquet_basis(
+                state00, taulist[0])
+
+        dims = np.shape(state00.full())
+
+        taulist_test = np.array(
+            taulist) % self.floquet_basis.T
+        taulist_correction = np.argwhere(abs(
+            taulist_test - self.floquet_basis.T) < 1e-10)
+        taulist_test_new = np.round(
+            taulist_test, 10)
+        taulist_test_new[taulist_correction] = 0
+
+        def list_duplicates(seq):
+            tally = {}
+            for i, item in enumerate(taulist_test_new):
+                try:
+                    tally[item].append(
+                        i)
```

Comment:
You have a lot of this strange line splitting...
When it fit on one line, it should not be split.